pulse-rs/src/util.rs

```diff
@@ -17,3 +18,24 @@ impl<'a, U> UnwrapCStr for U
         self.into().map(|o| o.as_ptr()).unwrap_or(0 as *const _)
     }
 }
+
+pub fn map_to_mut_ptr<T, U, F: FnOnce(&T) -> *mut U>(t: Option<&mut T>, f: F) -> *mut U {
+    match t {
+        Some(x) => f(x),
+        None => ptr::null_mut(),
+    }
+}
+
+pub fn str_to_ptr(s: Option<&CStr>) -> *const c_char {
+    match s {
+        Some(x) => x.as_ptr(),
+        None => ptr::null(),
+    }
+}
+
+pub fn to_ptr<T>(t: Option<&T>) -> *const T {
+    match t {
+        Some(x) => x as *const T,
+        None => ptr::null(),
+    }
+}
```

Comment:
The potential misuse of these utils worries me a bit, but I don't have a good alternative to offer right now, other than maybe doing without them so that the risk is more obvious at the points of use.